lib/internal/test_runner/mock/mock_timers.js

```diff
@@ -251,6 +358,9 @@ class MockTimers {
 
           nodeTimersPromises.setInterval = this.#realPromisifiedSetInterval;
         },
+        Date: () => {
+          globalThis.Date = this.#RealDate;
```

Comment:
If Date has been deleted, or made nonwritable, this would fail - i think it should cache the original descriptor instead of just the original value, and use ObjectDefineProperty here

Reply:
This also changes the behavior of the create date method, but that makes sense, I'll store the descriptor in the `#RealDate` and change the name to `#nativeDateDescriptor` then we can get the property back with `ObjectDefineProperty`